### PR TITLE
Update singleton-objects.md to use scala 2/3 tabs

### PR DESCRIPTION
**Pull request by Sporarum**

### CODE REVIEW

File: _tour/singleton-objects.md [hunk 1]
```diff
@@ -69,10 +119,34 @@ val circle1 = Circle(5.0)
...
 circle1.area
 ```
+{% endtab %}
+
+{% tab 'Scala 3' for=companion-object-circle %}
+```scala
+import scala.math._
+
+case class Circle(radius: Double):
+  import Circle._
```
Comment (julienrf) [inline _tour/singleton-objects.md:129]:
```suggestion
  import Circle.*
```

File: _tour/singleton-objects.md [hunk 2]
```diff
@@ -69,10 +119,34 @@ val circle1 = Circle(5.0)
 
 circle1.area
 ```
+{% endtab %}
+
+{% tab 'Scala 3' for=companion-object-circle %}
+```scala
+import scala.math._
```
Comment (julienrf) [inline _tour/singleton-objects.md:126]:
```suggestion
import scala.math.*
```

Comment (julienrf) [reply _tour/singleton-objects.md:126]: Or even:

```suggestion
import scala.math.pow
```

(and update the Scala 2 snippet accordingly)

File: _tour/singleton-objects.md [hunk 3]
```diff
@@ -95,6 +169,31 @@ scalaCenterEmail match {
   case None => println("Error: could not parse email")
 }
 ```
+{% endtab %}
+
+{% tab 'Scala 3' for=companion-object-email %}
+```scala mdoc
```
Comment (julienrf) [inline _tour/singleton-objects.md:175]:
~~~suggestion
```scala
~~~

We can’t use mdoc on Scala 3 snippets.

File: _tour/singleton-objects.md [hunk 4]
```diff
@@ -95,6 +169,31 @@ scalaCenterEmail match {
...
+class Email(val username: String, val domainName: String)
+
+object Email:
+  def fromString(emailString: String): Option[Email] = 
+    emailString.split('@') match
+      case Array(a, b) => Some(new Email(a, b))
```
Comment (julienrf) [inline _tour/singleton-objects.md:181]:
```suggestion
      case Array(a, b) => Some(Email(a, b))
```

**Review by julienrf — CHANGES_REQUESTED** [review]
Thank you @Sporarum! I’ve noticed a couple of things that could be improved.

**Comment by Sporarum** [comment]
Thanks @julienrf, I added these changes !

**Review by julienrf — APPROVED** [review]

**Pull request merged**